abdiff/core/collate_ab_transforms.py

```diff
@@ -216,24 +215,51 @@ def validate_output(dataset_path: str) -> None:
     """Validate the output of collate_ab_transforms.
 
     This function checks whether the collated dataset is empty
-    and whether any or both 'record_*' columns are empty.
+    and whether any or both 'record_a' or 'record_b' columns are
+    totally empty.
     """
-    collated_dataset = ds.dataset(dataset_path)
-    collated_table = collated_dataset.to_table(columns=["record_a", "record_b"])
+    collated_parquet_glob = f"{dataset_path}/**/*.parquet"
+
+    with duckdb.connect(":memory:") as con:
+        # check if the table is empty
+        record_count = con.execute(
+            "SELECT COUNT(*) FROM read_parquet($collated_parquet_glob)",
```

Comment:
Commenting here, but applies to pattern of the queries.

Another option when we have a DuckDB connection, assuming we reuse it for multiple queries, is that we could create a view that we reuse in other queries, e.g.

```python
con.execute(
    """
    CREATE VIEW collated as 
    SELECT * FROM read_parquet($collated_parquet_glob,hive_partitioning=true)
    """,
    {"collated_parquet_glob": collated_parquet_glob},
)
# no return value needed, as it creates a database object used in future queries
```

With that view created, which lasts for the duration of the connection, you could write queries like the following, no longer needing to repeat the `read_parquet(...)`:

```sql
SELECT COUNT(*) FROM collated;
```

100% optional.  Approving regardless of change, and no need for review if changed!

Reply:
@ehanson8 @ghukill As discussed:
* Updated validation function to create a view of the collated table that gets used by the subsequent queries.

Also worth noting: 
* DuckDB does not support using prepared statements when creating views (seems to be a [known issue](https://github.com/duckdb/duckdb/issues/13069))
* Can't use parameters in place of column names in query